### ISSUE TITLE
Windows/TUI: Printing to stdout during initialization prevents starting normally

### ISSUE THREAD

**Issue by eltechnic0** [neovim/neovim#7967]
- `nvim --version`: NVIM v0.2.2 (also with very recent 0.2.3-580-gec62f925a)
- Vim (version: ) behaves differently? No
- Operating system/version: Windows 10 Pro 1709
- Terminal name/version: cmd.exe
- `$TERM`: empty, cygwin (when running cmd.exe under cmder)

### Steps to reproduce

```
nvim -u init.test.vim
```

with `init.test.vim` being:

```
echom "Test echom"
```

### Actual behaviour

The following gets printed to the terminal:

```
Test echom
Press ENTER or type command to continue
```

And nothing else happens no matter what keys you press. Only `ctrl-c` gets you out to the terminal, and then everything you typed gets printed.

This seems to happen when anything prints messages during initialization.

I am able to start normally even with a complex init file only if no messages get printed.

### Expected behaviour

Display the message and start normally.

**Comment by janlazo** [comment]
```dosbatch
nvim -u NONE --cmd "echomsg 'Hello'"
```

is enough to reproduce this bug.

**Comment by DancingQuanta** [comment]
I have observed this as I tried to load my sizable vimrc configuration and it expectly complains about bad python interpreter and became uninteractive.

**Comment by janlazo** [comment]
@DancingQuanta If nvim outputs any error during startup, then it gets stuck (or the screen for the buffer doesn't show). This applies to `nvim --embed` for GUIs. Use `silent!` as a workaround for now and, worst case, don't use your python config for neovim on Windows.

**Comment by DancingQuanta** [comment]
Yes, I have to use a slimmed down config to continue working. I was trying this out as a potential replacement for cygwin nvim. But this windows nvim is not ready as par the freezing problem.

**Comment by erw7** [comment]
This issue has been fixed by #9884. It seems to pass the test of this problem added in  https://github.com/neovim/neovim/pull/10098/commits/33e0d00cd64a89c966e7b191dee3ac4e6db66b11. Therefore, I think that it is good to close this issue.

**Comment by DancingQuanta** [comment]
I seem to be experiencing issues as OP on latest neovim.

* nvim --version: NVIM v0.3.8 (install with 'scoop')
* Vim (version:  8.1 (2018 May 18, compiled Jul 21 2019 22:02:42)) behaves differently? No
* Operating system/version: Microsoft Windows 10 Enterprise Version 10.0.17134 Build 17134
* Terminal name/version: cmd.exe (Version 10.0.17134.167]
* $TERM: empty

# Reproducible situation

In `cmd.exe`
```
nvim -u NONE --cmd "echomsg 'Hello'"
```
yields the message `Press ENTER or type command to continue` and I have to execute `ctrl+c` move pass that message.

Is #9884 included in the 0.3.8 version of neovim?

**Comment by justinmk** [comment]
> Is #9884 included in the 0.3.8 version of neovim?

No.
